File: 0797-all-paths-from-source-to-target/0797-all-paths-from-source-to-target.py
class Solution:
    def allPathsSourceTarget(self, graph: List[List[int]]) -> List[List[int]]:
        end = len(graph) - 1
        ans = []

        def dfs(start, path):
            if start == end:
                ans.append(path[:])
                return

            for neighbor in graph[start]:
                path.append(neighbor)
                dfs(neighbor, path)
                path.pop()

        dfs(0, [0])

        return ans

File: 0797-all-paths-from-source-to-target/0797-all-paths-from-source-to-target.js
/**
 * @param {number[][]} graph
 * @return {number[][]}
 */
var allPathsSourceTarget = function (graph) {
    const n = graph.length;
    const destination = n - 1;
    const ans = [];

    const dfs = function (source, path) {
        if (source === destination) {
            ans.push([...path]);
            return;
        }

        for (const neighbor of graph[source]) {
            path.push(neighbor);
            neighbor && dfs(neighbor, path);
            path.pop();
        }
    }

    dfs(0, [0]);

    return ans;
};